app/email-incoming/index.ts

```diff
@@ -0,0 +1,180 @@
+
+import type { Source } from 'mailparser'
+import { simpleParser } from 'mailparser'
+
+import {
+  emailIsAutoReply,
+  subjectIsValid,
+  formatAuthor,
+} from '../routes/circulars/circulars.lib'
+
+import { tables } from '@architect/functions'
+import { extractAttributeRequired, extractAttribute } from '~/lib/cognito'
+
+const s3 = new S3Client({})
+const client = new CognitoIdentityProviderClient({})
+const sesClient = new SESClient({})
+const doc = DynamoDBDocument.from(new DynamoDBClient({}))
+
+const USER_POOL_ID = process.env.USER_POOL_ID
+
+const domain =
+  process.env.NODE_ENV === 'development'
+    ? 'dev.'
+    : process.env.NODE_ENV === 'test'
+    ? 'test.'
+    : ''
+
+export async function handler(event: S3Event) {
+  const getObjectParamPromises = event.Records.filter(
+    (record) => record.eventName == 'ObjectCreated:Put'
+  ).map((record) => {
+    return s3.send(
+      new GetObjectCommand({
+        Bucket: record.s3.bucket.name,
+        Key: decodeURIComponent(record.s3.object.key.replace(/\+/g, ' ')),
+      })
+    )
+  })
+
+  // Get data from S3
+  const objects = await Promise.all(getObjectParamPromises)
+
+  objects.forEach(async (object) => {
+    if (!object.Body) throw new Error('S3 object has no body')
+
+    const parsed = await simpleParser(object.Body as Source)
+
+    if (!parsed.from) throw new Error('S3 object has no sender')
+
+    const userEmail = parsed.from.value[0].address
+    if (!userEmail)
+      throw new Error(
+        'Error parsing sender email from model: ' + JSON.stringify(parsed.from)
+      )
+
+    if (!parsed.subject || !parsed.text) {
+      await sendEmail(
+        userEmail,
+        'Invalid GCN Circular Submission',
+        'Subject and Body are required to submit a GCN Circular via email.'
+      )
+      return
+    }
+
+    // Check if submitter is in the submitters group
+    var listUsersParams = {
+      GroupName: 'gcn.nasa.gov/circular-submitter',
+      UserPoolId: USER_POOL_ID,
+    }
+
+    const data = await client.send(new ListUsersInGroupCommand(listUsersParams))
+
+    const userData = data.Users?.filter(
+      (user) => extractAttributeRequired(user, 'email') == userEmail
+    ).map((user) => ({
+      sub: extractAttributeRequired(user, 'sub'),
+      email: extractAttributeRequired(user, 'email'),
+      name: extractAttribute(user, 'name'),
+      affiliation: extractAttribute(user, 'custom:affiliation'),
+    }))
+
+    if (!userData || userData.length == 0) {
+      await sendEmail(
+        userEmail,
+        'GCN Circular Submission Warning: Missing permissions',
+        'You do not have the required permissions to submit GCN Circulars. If you believe this to be a mistake, please fill out the form at https://heasarc.gsfc.nasa.gov/cgi-bin/Feedback?selected=kafkagcn, and we will look into resolving it as soon as possible.'
+      )
+      return
+    }
+
+    // Validate Subject Line
+    if (emailIsAutoReply(parsed.subject)) {
```

Comment:
Incorporate `emailIsAutoReply` into `subjectIsValid` so that you only have to call `subjectIsValid`.